### PR TITLE
Improve season support for UnionFansub

### PR DESCRIPTION
**Pull request by MiguelAngelLV**
#### Description

Improve season support for Union Fansub inspired by Nya indexer and add release group (fansub) to title name.

### CODE REVIEW

**Review by garfield69 — CHANGES_REQUESTED** [review]
something is not quite right.
seeing some rows returning
```
CardigannIndexer (unionfansub): Error while parsing row '
<tr>
	<td align="center" style="padding: 0px">
		<a href="browse.php?cat=1">
			<img border="0" src="./pic/caticons/cat_anime.png" alt="Anime" title="Anime">
		</a>
	</td>
	<td align="left">
		<div>
			<a href="details.php?id=(removed)&amp;hit=1">
				<b class="name">Mushoku Tensei: Isekai Ittara Honki Dasu</b>
			</a>
			<span class="fansub">Hakoniwa</span>
		</div>
	</td>
	<td align="center">
		<span class="icon jap" title="Japones"></span>
	</td>
	<td align="center">
		<span class="icon lat" title="Latino"></span>
	</td>
	<td align="center">
		<span style="white-space: nowrap;"></span>
	</td>
	<td align="right">
		<b>
			<a href="filelist.php?id=(removed)">11</a>
		</b>
	</td>
	<td align="right">0</td>
	<td align="center">
		<span style="white-space: nowrap;">175w5d</span>
	</td>
	<td align="center">7,62 GB</td>
	<td align="center">
		<a href="snatches.php?id=(removed)">255</a>
	</td>
	<td align="right">
		<b>
			<a href="peerlist.php?id=(removed)">
				<font color="#000000">10</font>
			</a>
		</b>
	</td>
	<td align="right">0</td>
	<td align="center">
		<a href="userdetails.php?id=(removed)">
			<s>Kaizer</s>
		</a>
	</td>
</tr>':

System.Exception: Error while parsing field=title_fansub, selector=td:nth-child(2) .fansubbd, value=<null>: Selector "td:nth-child(2) .fansubbd" didn't match 
```

File: src/Jackett.Common/Definitions/unionfansub.yml [hunk 1]
```diff
@@ -98,8 +98,21 @@ search:
       filters:
         - name: querystring
           args: cat
-    title:
+    title_fansub:
+      selector: td:nth-child(2) .fansubbd
```
Comment (garfield69) [inline src/Jackett.Common/Definitions/unionfansub.yml:102]:
```suggestion
      selector: td:nth-child(2) .fansubbd
      optional: true
```

File: src/Jackett.Common/Definitions/unionfansub.yml [hunk 2]
```diff
@@ -98,8 +98,21 @@ search:
...
+        - name: re_replace
+          args: ["(?i)\\b\\d{1,4}[ºª]\\b", ""]
+        - name: re_replace
+          args: ["(?i)\\b(Temporada|Season) (\\d+)\\b", "S$2"]
+        - name: re_replace
+          args: ["(?i)\\b(\\d+)(st|nd|rd|th) (Season|Temporada)\\b", "S$1"]
+        - name: re_replace
+          args: ["(?i)\\b(Season|Temporada) (\\d+)(st|nd|rd|th)\\b", "S$2"]
+    title:
+      text: "{{ .Result.title_season }} [{{ .Result.title_fansub }}]"
```
Comment (garfield69) [inline src/Jackett.Common/Definitions/unionfansub.yml:115]:
```suggestion
      text: "{{ .Result.title_season }} {{ if .Result.title_fansub }} [{{ .Result.title_fansub }}]{{ else }}{{ end }}"
```

**Review by garfield69 — CHANGES_REQUESTED** [review]

**Review by garfield69 — CHANGES_REQUESTED** [review]
what is the intended outcome for `Loonatics Unleashed (Los Lunáticos) - Temporada 1ª` ?
currently getting `Loonatics Unleashed (Los Lunáticos) - Temporada`

**Comment by MiguelAngelLV** [comment]
Um... It should return Los Lunaticos S1... I will check it

**Comment by MiguelAngelLV** [comment]
Fixed

image

**Review by garfield69 — APPROVED** [review]
LGTM

**Comment by garfield69** [comment]
v0.22.468

**Pull request merged**